### PR TITLE
Post release KFS implementation cleanup

### PR DESCRIPTION
**Pull request by atobiszei**
* Move KFS/TFS /gRPCutils out out precision.hpp & tensorinfo.hpp
* Make OVMS use just one using directive with KFS gRPC structures
* Increase code coverage in PipelineDefinitionStatus

JIRA:CVS-92107

### CODE REVIEW

File: src/dlnodesession.cpp [hunk 1]
```diff
@@ -108,47 +106,22 @@ Status DLNodeSession::prepareInputsAndModelForInference() {
...
         }
-    } else if (requestedBatchSize.has_value()) {
-        auto status = this->model->reloadModel(requestedBatchSize, {}, this->modelUnloadGuard);
-        if (!status.ok()) {
+        if (status == StatusCode::INVALID_SHAPE) {
             return status;
         }
     }
-    return StatusCode::OK;
+    return status;
```
Comment (dkalinowski) [inline src/dlnodesession.cpp:124]: You change behavior here, previously if there was some kind of error in for loop it was not returned in 151, now it is. Should we add test for that?

File: src/gatherexitnodeinputhandler.hpp [hunk 2]
```diff
@@ -26,19 +26,18 @@
...
+#include "kfs_grpc_inference_service.hpp"
 #include "logging.hpp"
 #include "profiler.hpp"
 #include "status.hpp"
 
 namespace ovms {
 
 Status prepareConsolidatedTensorImpl(tensorflow::serving::PredictResponse* response, char*& tensorOut, const std::string& name, size_t size);
-Status prepareConsolidatedTensorImpl(::inference::ModelInferResponse* response, char*& tensorOut, const std::string& name, size_t size);
+Status prepareConsolidatedTensorImpl(KFSResponse* response, char*& tensorOut, const std::string& name, size_t size);
```
Comment (dkalinowski) [inline src/gatherexitnodeinputhandler.hpp:40]: We have moved implementation from cpp unit, why should we keep prototype here?

File: src/dlnodesession.cpp (hunk 1 above)
Comment (atobiszei) [reply src/dlnodesession.cpp:124]: This removes support for bs=auto/shape=auto from pipeline execution flow. This support was disabled anyway in configuration validation.

File: src/gatherexitnodeinputhandler.hpp (hunk 2 above)
Comment (atobiszei) [reply src/gatherexitnodeinputhandler.hpp]: Done

File: src/gatherexitnodeinputhandler.hpp [hunk 3]
```diff
@@ -26,19 +26,17 @@
...
 
+#include "kfs_frontend/kfs_utils.hpp"
+#include "kfs_grpc_inference_service.hpp"
 #include "logging.hpp"
 #include "profiler.hpp"
 #include "status.hpp"
 
 namespace ovms {
 
 Status prepareConsolidatedTensorImpl(tensorflow::serving::PredictResponse* response, char*& tensorOut, const std::string& name, size_t size);
```
Comment (dkalinowski) [inline src/gatherexitnodeinputhandler.hpp:39]: We untie KFS from gatherexitnodeinputhandler. We will end up with both prepare functions in separate places. Maybe move TFS one as well?

File: src/kfs_frontend/kfs_utils.hpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+#include <string>
+
+#include "../kfs_grpc_inference_service.hpp"
```
Comment (dkalinowski) [inline src/kfs_frontend/kfs_utils.hpp:19]: Shouldnt this file be inside kfs_frontend directory as well?

**Review by dkalinowski — APPROVED** [review]

**Review by michalkulakowski — APPROVED** [review]

**Pull request merged**